gpt_index/indices/query/base.py

```diff
@@ -129,6 +131,18 @@ def _should_use_node(self, node: Node) -> bool:
                 if w in words:
                     return False
 
+        if (
+            similarity_tracker is not None
+            and similarity_tracker.similarity_cutoff is not None
+        ):
+
+            similarity = similarity_tracker.find(node)
+            if (
+                similarity is not None
```

Comment:
what if the similarity is None? we probably shouldn't include that either right? (or raise an error because its an edge case)

Reply:
hmm i thought i had found when i ran it in the past that similarity was occasionally None unexpectedly but unable to repro now, will make the change and see if tests break

Reply:
actually sorry i take this comment back. I noticed that similaritytracker adds similarities lazily so raising an error if similarity is None might break stuff. Maybe we can just return False is similarity is None (so the node won't get included) for now, and I can think about a better approach later